### PR TITLE
fix: Add isFirst check before err check

### PR DESCRIPTION
**Pull request by kylecarbs**
This was causing TestBlockNonBrowser to hang and fail.

### CODE REVIEW

**Comment by kylecarbs** [comment]
Force merging this because CI is in shambles right now!

File: codersdk/workspaceagents.go [hunk 1]
```diff
@@ -331,21 +331,21 @@ func (c *Client) DialWorkspaceAgentTailnet(ctx context.Context, logger slog.Logg
...
-			if err != nil {
-				if errors.Is(err, context.Canceled) {
-					return
-				}
-				logger.Debug(ctx, "failed to dial", slog.Error(err))
-				continue
-			}
 			if isFirst {
-				if res.StatusCode == http.StatusConflict {
+				if err != nil && res.StatusCode == http.StatusConflict {
```
Comment (mafredri) [inline codersdk/workspaceagents.go:335]: Should this be `err == nil`? Thinking this will still have the potential nil pointer deref if `err != nil`?

Comment (kylecarbs) [reply codersdk/workspaceagents.go:335]: Ahh, you're correct. I'll fix this now in another PR. I merged due to panic of `main` being busted!

Comment (kylecarbs) [reply codersdk/workspaceagents.go:335]: It seems the error isn't nil there actually, so a weird case. I'm just going to check for the status code, then it should be fine!

Comment (kylecarbs) [reply codersdk/workspaceagents.go:335]: https://github.com/coder/coder/pull/4327

**Pull request merged**